cargo-shuttle/src/lib.rs

```diff
@@ -1042,7 +1051,7 @@ impl Shuttle {
         )> = Vec::new();
 
         if !portpicker::is_free_tcp(run_args.port) {
-            Shuttle::find_available_port(&mut run_args);
+            Shuttle::find_available_port(&mut run_args, services.len());
         }
```

Comment:
We should let the function do all the port checking, since it needs to check multiple ports.